### ISSUE TITLE
Exception when applying forward-pass to Mask-RCNN Inception v2 Model

### ISSUE THREAD

**Issue by Razor1O9** [opencv/opencv#16305]
### System information (version)
- OpenCV => 4.1.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Qt Qreator

**Model:** mask_rcnn_inception_v2_coco; Version: 2018_01_28
**Train Config:** mask_rcnn_inception_v2_coco.config

**Code:**
```

Net ub6net = readNetFromTensorflow(model, config);
ub6net.setPreferableBackend(DNN_BACKEND_OPENCV);
ub6net.setPreferableTarget(DNN_TARGET_CPU);

blobFromImage(frame,blob, 1.0, size(frame.cols, frame.rows), Scalar(), true, false);

ub6net.setInput(blob);

vector<String> outputNames(2);
outputNames[0] = "detection_final_out";
outputNames[1] = "detection_mask";

vector<Mat> output;
ub6net.forward(output, outputNames); // **EXCEPTION GETS THROWN HERE**
```

____________________________________________

**Exception:**
dnn.cpp:525: (-2:Unspecified error) Can't create layer "FirstStageFeatureExtractor/InceptionV2/InceptionV2/Conv2d_1a_7x7/BatchNorm/FusedBatchNormV3" of type "FusedBatchNormV3" in function 'getLayerInstance'.

____________________________________________

I've used the GraphTransformation Tool to try and get rid of the Exception.
I've applied the following Graph transformations:

sort_by_execution_order
rename_op(old_op_name=FusedBatchNormV3, new_op_name=FusedBatchNorm)
fold_constants(ignore_errors = true)
fold_old_batch_norms
fold_batch_norms
remove_nodes(op=Identity, op=CheckNumerics)

**The Exception than changes to:**
dnn.cpp:1295: error: (-204:Requested object was not found) Layer with requested id=-1 not found in function 'getLayerData' 

**I've uploaded all important files (model, train-config) to Mega:**
https://mega.nz/fm/Rh1EAA5D

**Comment by dkurt** [comment]
Looks kike we need a PR similar to this one: https://github.com/opencv/opencv/pull/15986 for Faster and Mask R-CNNs.

**Comment by Razor1O9** [comment]
**Greetings,**

I've tried to apply the changes from #15986 to the tf_text_graph_mask_rcnn.py by adding 'FusedBatchNorm' inside scopesToKeep. However, the described behaviour has not changed yet. Do I have to add the paramter somewhere else inside the script as well?

I already renamed the op descriptor from FusedBatchNormV3 to FusedBatchNorm, as mentioned in my first comment on this issue.

### Changes made to tf_text_graph_mask_rcnn.py:

**Line 13**
scopesToKeep = ('...' , 'FusedBatchNorm', '...')

**Comment by dkurt** [comment]
@Razor1O9, please review https://github.com/opencv/opencv/pull/15986/files more carefully. You need to add `FusedBatchNormV3` but not `FusedBatchNorm`

**Comment by Razor1O9** [comment]
> 
> 
> @Razor1O9, please review https://github.com/opencv/opencv/pull/15986/files more carefully. You need to add `FusedBatchNormV3` but not `FusedBatchNorm`

I applied graph transformation to rename all the FusedBatchNormV3 layers inside my model to FusedBatchNorm **before** using the tf_text_graph_mask_rcnn.py script. Does this not change the approach destribed in #15986? Sorry for the confusion.

**Comment by Razor1O9** [comment]
**Solved:** 
I had to replace all mentions of FusedBatchNormV3 inside the name attribute of all nodes. 
It was not enough to replace the op name.

**Example:**

name: "FeatureExtractor/MobileNetV2/expanded_conv_2/project/BatchNorm/**FusedBatchNormV3**"
op: "FusedBatchNorm" // this has the right name, due to the graph transformation mentioned below

The bold names have to be replaced, this solves the exception.

Additionally I added "FusedBatchNorm" to the scopesToKeep in tf_text_graph_mask_rcnn.py and used the graph transformation tool to rename all FusedBatchNormV3 ops of the model.pb to FusedBatchNorm before running the script.
